arch/cortex-m/samv71-hal/src/hal.rs

```diff
@@ -0,0 +1,56 @@
+//! System HAL implementation for x86 target.
+
+use aerugo_hal::system_hal::{SystemHal, SystemHardwareConfig};
+use bare_metal::CriticalSection;
+
+use crate::peripherals::Peripherals;
+
+/// HAL implementation for x86.
+pub struct Hal {
+    /// Hardware peripherals.
+    _peripherals: Peripherals,
+}
+
+impl Hal {
+    /// Frequency for the time types (TODO)
+    const TIMER_FREQ: u32 = 1_000_000;
+
+    /// Create new HAL instance.
+    pub const fn new(peripherals: Peripherals) -> Self {
+        Hal {
+            _peripherals: peripherals,
+        }
+    }
+}
+
+impl SystemHal for Hal {
+    type Instant = crate::time::TimerInstantU64<{ Hal::TIMER_FREQ }>;
+    type Duration = crate::time::TimerDurationU64<{ Hal::TIMER_FREQ }>;
+
+    fn configure_hardware(&mut self, _config: SystemHardwareConfig) {
```

Comment:
I assume  master clock is part of the platform-specific SystemHardwareConfig?

Reply:
That is the plan, yes.